compiler/qsc_project/src/manifest.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::Error;
+use serde::Deserialize;
+use std::path::PathBuf;
+use std::{
+    env::current_dir,
+    fs::{self, DirEntry, FileType},
+};
+
+pub const MANIFEST_FILE_NAME: &str = "qsharp.json";
+
+/// A Q# manifest, used to describe project metadata.
+#[derive(Deserialize, Debug, Default)]
+pub struct Manifest {
+    pub author: Option<String>,
+    pub license: Option<String>,
+    #[serde(default)]
+    pub exclude_files: Vec<String>,
+}
+
+/// Describes the contents and location of a Q# manifest file.
+#[derive(Debug)]
+pub struct ManifestDescriptor {
+    pub manifest: Manifest,
+    pub manifest_dir: PathBuf,
+}
+
+impl ManifestDescriptor {
+    pub(crate) fn new(manifest: Manifest, manifest_dir: PathBuf) -> Self {
+        Self {
+            manifest,
+            manifest_dir,
+        }
+    }
+}
+
+impl Manifest {
+    /// Starting from the current directory, traverse ancestors until
+    /// a manifest is found.
+    /// Returns an error if there are any filesystem errors, or if
+    /// a manifest file exists but is the wrong format.
+    /// Returns `Ok(None)` if there is no file matching the manifest file
+    /// name.
+    pub fn load() -> std::result::Result<Option<ManifestDescriptor>, Error> {
+        let current_dir = current_dir()?;
```

Comment:
What does 'current_dir' even mean in the context of wasm or the browser? Looking at the implementation, it appears to call os_imp::getcwd(), so I'm guess it ain't gonna work (https://doc.rust-lang.org/src/std/env.rs.html#56)

Reply:
This module isn't conditional being the 'fs' feature if I'm reading correctly, so it's always compiled in - is that right?. Why would you want to put a 'default fs-based implemention' in this project if it isn't usable in any of our supported scenarios? (I'm guessing Jupyter Notebooks would usually want to point to the manifest/dir also, and not just use current directory).

I don't think we should have default implementation of what the host should provide in the project, especially if most of our scenarios won't use it.